language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddIsolatedQualifierCodeAction.java

```diff
@@ -68,56 +75,90 @@ public boolean validate(Diagnostic diagnostic,
+            return Collections.emptyList();
         }
-        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(nonTerminalNode);
+
+        // Obtain the symbol of the referred function
+        Optional<Symbol> symbol = getReferredSymbol(context, nonTerminalNode);
         if (symbol.isEmpty() || symbol.get().getModule().isEmpty()) {
             return Collections.emptyList();
         }
 
+        // Obtain the current project
         Optional<Project> project = context.workspace().project(context.filePath());
         if (project.isEmpty()) {
             return Collections.emptyList();
         }
+
+        // Obtain the file path of the referred symbol
         Optional<Path> filePath = PathUtil.getFilePathForSymbol(symbol.get(), project.get(), context);
         if (filePath.isEmpty() || context.workspace().syntaxTree(filePath.get()).isEmpty()) {
             return Collections.emptyList();
         }
 
+        // Obtain the node of the referred symbol
         Optional<NonTerminalNode> node = CommonUtil.findNode(symbol.get(),
                 context.workspace().syntaxTree(filePath.get()).get());
-        if (node.isEmpty() || !node.get().kind().equals(SyntaxKind.FUNCTION_DEFINITION)) {
+        if (node.isEmpty() || isUnsupportedSyntaxKind(node.get().kind())) {
             return Collections.emptyList();
         }
-
         FunctionDefinitionNode functionDefinitionNode = (FunctionDefinitionNode) node.get();
-        Position position = PositionUtil.toPosition(functionDefinitionNode.functionKeyword().lineRange().startLine());
 
-        Range range = new Range(position, position);
-        String editText = SyntaxKind.ISOLATED_KEYWORD.stringValue() + " ";
-        TextEdit textEdit = new TextEdit(range, editText);
-        List<TextEdit> editList = List.of(textEdit);
-        String commandTitle = String.format(CommandConstants.MAKE_FUNCTION_ISOLATE, symbol.get().getName().orElse(""));
-        return Collections.singletonList(CodeActionUtil
-                .createCodeAction(commandTitle, editList, filePath.get().toUri().toString(), CodeActionKind.QuickFix));
+        return getCodeAction(context, functionDefinitionNode.functionKeyword(), symbol.get().getName().orElse(""),
+                filePath.get().toUri().toString());
     }
 
-    private List<CodeAction> getExplicitAnonFuncExpressionCodeAction(ExplicitAnonymousFunctionExpressionNode node,
-                                                                     CodeActionContext context) {
-        Position position = PositionUtil.toPosition(node.functionKeyword().lineRange().startLine());
-        Range range = new Range(position, position);
+    private static Optional<Symbol> getReferredSymbol(CodeActionContext context, NonTerminalNode node) {
+        SyntaxKind kind = node.kind();
+        if (kind == SyntaxKind.EXPLICIT_NEW_EXPRESSION || kind == SyntaxKind.IMPLICIT_NEW_EXPRESSION) {
+            try {
```

Comment:
Why do we need this try-catch block?

Reply:
All the exit paths of the instructions within the block follow the same recovery logic. This approach is more convenient and readable than having to check at each line.